src/Dev/State/FixtureTestState.php

```diff
@@ -0,0 +1,229 @@
+     */
+    protected function getFixturePaths($fixtures, SapphireTest $test)
+    {
+        return array_map(function ($fixtureFilePath) use ($test) {
+            return $this->resolveFixturePath($fixtureFilePath, $test);
+        }, $fixtures);
+    }
+
+    /**
+     * @param SapphireTest $test
+     */
+    protected function loadFixtures(SapphireTest $test)
+    {
+        $fixtures = $test::get_fixture_file();
+        $fixtures = is_array($fixtures) ? $fixtures : [$fixtures];
+        $paths = $this->getFixturePaths($fixtures, $test);
+        foreach ($paths as $fixtureFile) {
+            $this->loadFixture($fixtureFile, $test);
+        }
+    }
+
+    /**
+     * @param string $fixtureFile
+     * @param SapphireTest $test
+     */
+    protected function loadFixture($fixtureFile, SapphireTest $test)
+    {
+        /** @var YamlFixture $fixture */
+        $fixture = Injector::inst()->create(YamlFixture::class, $fixtureFile);
+        $fixture->writeInto($this->getFixtureFactory(get_class($test)));
+    }
+
+    /**
+     * Map a fixture path to a physical file
+     *
+     * @param string $fixtureFilePath
+     * @param SapphireTest $test
+     *
+     * @return string
+     */
+    protected function resolveFixturePath($fixtureFilePath, SapphireTest $test)
+    {
+        // Support fixture paths relative to the test class, rather than relative to webroot
+        // String checking is faster than file_exists() calls.
+        $isRelativeToFile
+            = (strpos('/', $fixtureFilePath) === false)
+            || preg_match('/^(\.){1,2}/', $fixtureFilePath);
+
+        if ($isRelativeToFile) {
+            $resolvedPath = realpath($this->getTestAbsolutePath($test) . '/' . $fixtureFilePath);
+            if ($resolvedPath) {
+                return $resolvedPath;
+            }
+        }
+
+        // Check if file exists relative to base dir
+        $resolvedPath = realpath(Director::baseFolder() . '/' . $fixtureFilePath);
+        if ($resolvedPath) {
+            return $resolvedPath;
+        }
+
+        return $fixtureFilePath;
+    }
+
+    /**
+     * Useful for writing unit tests without hardcoding folder structures.
+     *
+     * @param SapphireTest $test
+     *
+     * @return string Absolute path to current class.
+     */
+    protected function getTestAbsolutePath(SapphireTest $test)
+    {
+        $filename = ClassLoader::inst()->getItemPath(get_class($test));
+        if (!$filename) {
+            throw new LogicException('getItemPath returned null for ' . static::class
+                . '. Try adding flush=1 to the test run.');
+        }
+        return dirname($filename);
+    }
+
+    /**
+     * @param SapphireTest $test
+     *
+     * @return bool
+     */
+    protected function testNeedsDB(SapphireTest $test)
+    {
+        $annotations = $test->getAnnotations();
+
+        // annotation explicitly disables the DB
+        if (array_key_exists('useDatabase', $annotations['method'])
```

Comment:
I never even knew this was a thing.

Reply:
I saw the PR adding it a while ago - we really should add the annotations to appropriate tests in core tests :)